src/Lykke.AzureStorage/Tables/Entity/Metamodel/EntityMetamodelImpl.cs

```diff
@@ -23,12 +26,91 @@ public IStorageValueSerializer TryGetSerializer(PropertyInfo property)
                 // Serializer defined on the property, overrides that one, defined on the type, which property has
 
                 return _provider.TryGetPropertySerializer(property) ??
-                       _provider.TryGetTypeSerializer(property.PropertyType);
+                       _provider.TryGetTypeSerializer(property.PropertyType) ??
+                       TryGetCollectionSerializer(property.PropertyType) ??
+                       TryGetNullableSerializer(property.PropertyType);
+
             }
             catch (Exception ex)
             {
                 throw new InvalidOperationException($"Failed to get storage value serializer for property {property.DeclaringType}.{property.Name}, in context of type {property.ReflectedType}", ex);
             }
         }
+
+        public ValueTypeMergingStrategy? TryGetValueTypeMergingStrategy(Type type)
+        {
+            try
+            {
+                return _provider.TryGetValueTypeMergingStrategy(type);
+            }
+            catch (Exception ex)
+            {
+                throw new InvalidOperationException($"Failed to get value type merging strategy factory for entity type {type}", ex);
+            }
+        }
+
+        private IStorageValueSerializer TryGetCollectionSerializer(Type collectionType)
+        {
+            var enumerbleGenericArguments = GetGenericArgumentsOfAssignableType(collectionType, typeof(IEnumerable<>));
+            if (enumerbleGenericArguments != null)
+            {
+                var elementType = enumerbleGenericArguments.First();
```

Comment:
Use indexer

Reply:
Why? `First()` is more semantic, performance of the method is not critical, since result is cached. moreover First() implementation takes care about Array and actually do exactly the `array[0]` - https://github.com/dotnet/corefx/blob/master/src/System.Linq/src/System/Linq/First.cs#L51